clang/lib/AST/DeclCXX.cpp

```diff
@@ -3423,6 +3425,30 @@ VarDecl *BindingDecl::getHoldingVar() const {
   return VD;
 }
 
+void DecompositionDecl::VisitHoldingVars(
+    llvm::function_ref<void(VarDecl *)> F) const {
+  VisitBindings([&](BindingDecl *BD) {
+    if (VarDecl *VD = BD->getHoldingVar())
+      F(VD);
+  });
+}
+
+void DecompositionDecl::VisitBindings(
```

Comment:
This one is a bit more complicated?  But I'd be surprised if there isn't an existing llvm iterator that can do this.